packages/flame/lib/src/value_cache.dart

```diff
@@ -0,0 +1,28 @@
+/// Used for caching calculated values, the cache is determined to be valid by
+/// comparing a list of values that can be of any type and is compared to the
+/// values that was last used when the cache was updated.
+class ValueCache<T> {
```

Comment:
We already have a cache class: https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/memory_cache.dart

But I think they are different things right? If so, maybe this deserves a folder of its own? So we can keep cache related sources together?

Reply:
Aah forgot about that one, but yes, they are definitely different since this one has premises regarding whether the cache is valid or not. Maybe it could extend `MemoryCache`, i'll look into it and create a directory for them!

Reply:
I think it'll be better to keep them separate, but I moved them to a common directory.
Do we want to expose these ones too? If so, which export file should they come with?

Reply:
I would export only if `MemoryCache` was already exported, which I think wasn't but I am not sure right now.

Reply:
It wasn't exported, let's keep them unexported then. I'll mark this discussion as resolved.